src/core/ComparisonBarChart.js

```diff
@@ -1,112 +1,151 @@
-  } = toReferenceProps(ref);
-  const groupColorScale = theme.group.colorScale;
-  const barProps = {
-    ...{
-      labels: datum => datum.y,
-      labelComponent: <VictoryLabel x={50} dy={-25} />
-    },
-    ...props
+    style: referenceStyleProp
+  } = toReferenceProps(referenceProp);
+  const referenceStyle = { ...chart.referenceStyle, ...referenceStyleProp };
+  const dataStyle = { ...chart.style.data, ...style.data };
+  const { colorScale } = chart;
+
+  const height = heightProp || chart.height;
+  const width = widthProp || chart.width;
+  const values = data.map(d => d.y).concat(referenceData.y);
+  const max = Math.max.apply(null, values);
+  const dataBarWidths = data.map(d => (d.y * width) / max);
+  const referenceDataBarWidth = (referenceData.y * width) / max;
+  const barHeight = barHeightProp || chart.barHeight;
+
+  const tooltip = (
+    <VictoryTooltip constrainToVisibleArea {...tooltipProps} theme={theme} />
+  );
+  const activateTooltip = (evt, newTooltipProps) => {
+    if (newTooltipProps && newTooltipProps.text) {
+      const { x: tipX, y: tipY } = Selection.getSVGEventCoordinates(evt);
+      setTooltipProps({ active: true, ...newTooltipProps, x: tipX, y: tipY });
+    }
   };
+
   return (
-    // The bar charts order is reversed, so the last will be at the top
-    <Chart theme={theme} horizontal={horizontal} width={width} height={height}>
-      {/* Legend */}
-      <VictoryBar
-        barWidth={5}
-        style={referenceStyle}
-        data={[referenceData]}
-        labels={({ datum }) => datum.y}
-        labelComponent={<VictoryLabel x={50} dy={-15} />}
-        {...props}
+    <CustomContainer
+      theme={theme}
+      horizontal={horizontal}
+      width={width}
+      height={height}
+    >
+      {dataBarWidths.map((barWidth, i) => (
+        <React.Fragment key={data[i].x}>
+          <VictoryLabel
+            capHeight={0}
+            dy={0}
+            lineHeight={0}
+            style={{ fill: colorScale[i % colorScale.length], ...dataStyle }}
+            text={data[i].y}
+            x={0}
+            y={(i + 1) * 40 + i * 10 - barHeight}
+          />
+          <Border
+            events={{
+              onMouseOver: evt =>
+                activateTooltip(evt, { text: `${data[i].x}: ${data[i].y}` }),
+              onMouseMove: evt =>
+                activateTooltip(evt, { text: `${data[i].x}: ${data[i].y}` }),
+              onMouseOut: () => setTooltipProps({ active: false })
+            }}
+            height={barHeight}
+            style={{ fill: colorScale[i % colorScale.length] }}
+            x={0}
+            width={barWidth}
+            y={(i + 1) * 40 + i * 10}
+          />
+        </React.Fragment>
+      ))}
+      <VictoryLabel
+        capHeight={0}
+        dy={0}
+        lineHeight={0}
+        style={referenceStyle.data}
+        text={referenceData.y}
+        x={0}
+        y={data.length * 40 + (data.length - 1) * 10 - 2 * barHeight + 70}
```

Comment:
I know we use alot of magic numbers, lets note this somewhere to start explaining any numbers we use and probably have them as static variables. 

This is for future work.